components/script/dom/abstractworkerglobalscope.rs

```diff
@@ -108,17 +108,17 @@ pub fn run_worker_event_loop<T, TimerMsg, WorkerMsg, Event>(
     let scope = worker_scope.upcast::<WorkerGlobalScope>();
     let timer_event_port = worker_scope.timer_event_port();
     let devtools_port = match scope.from_devtools_sender() {
-        Some(_) => Some(scope.from_devtools_receiver().select()),
+        Some(_) => Some(scope.from_devtools_receiver()),
         None => None,
     };
     let task_queue = worker_scope.task_queue();
     let event = select! {
-        recv(task_queue.select(), msg) => {
+        recv(task_queue.select()) -> msg => {
             task_queue.take_tasks(msg.unwrap());
             worker_scope.from_worker_msg(task_queue.recv().unwrap())
         },
-        recv(timer_event_port.select(), msg) => worker_scope.from_timer_msg(msg.unwrap()),
-        recv(devtools_port, msg) => worker_scope.from_devtools_msg(msg.unwrap()),
+        recv(timer_event_port) -> msg => worker_scope.from_timer_msg(msg.unwrap()),
+        // TODO: recv(devtools_port) -> msg => worker_scope.from_devtools_msg(msg.unwrap()),
```

Comment:
So here is the only thing I didn't know how to do. `select!` doesn't support `recv(Option<Receiver>)` anymore. Do I have to write that with `Select` then ?

Reply:
You could do that with `Select`, but you don't have to. There's another way:

```rust
recv(devtools_port.unwrap_or(&crossbeam_channel::never())) -> msg => ...
```

See [here](https://docs.rs/crossbeam-channel/0.3.0/crossbeam_channel/fn.never.html#examples) for documentation on `never`.

I removed the ability to use `Option<&Receiver<_>>` in `select!` because it was an ugly hack that abuses traits and doesn't work as well as I hoped for. Maybe I'll add it again sometime in the future when we get specialization in Rust.